### PR TITLE
Balance module refactoring

### PR DESCRIPTION
**Pull request by abdrasulov**

### CODE REVIEW

File: app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AuthManager.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    fun logout() {
+        walletManager?.clearWallets()
+        pinManager?.clear()
+        transactionManager?.clear()
+        localStorage.clearAll()
+
+//        secureStorage.clearAuthData()
```
[the remark this answers is not in the document]
Comment (abdrasulov) [reply app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AuthManager.kt]: We need to implement this method. I left it as a mark.

File: app/src/main/java/io/horizontalsystems/bankwallet/SingleLiveEvent.kt [hunk 2]
```diff
@@ -22,8 +22,17 @@ class SingleLiveEvent<T> : MutableLiveData<T>() {
...
     }
 
+    override fun postValue(value: T?) {
+        mPending.set(true)
+        super.postValue(value)
+    }
+
     fun call() {
         value = null
     }
```
Comment (rafaelekol) [inline app/src/main/java/io/horizontalsystems/bankwallet/SingleLiveEvent.kt]: why function is called 'callX'

File: app/src/main/java/io/horizontalsystems/bankwallet/core/factories/TransactionViewItemFactory.kt [hunk 3]
```diff
@@ -23,7 +22,7 @@ class TransactionViewItemFactory(
 
         val rateValue = when {
             record.rate != 0.0 -> record.rate
-            DateHelper.getSecondsAgo(record.timestamp * 1000) < latestRateFallbackThreshold * 60 -> rateManager.latestRates[record.coinCode]?.get(currencyManager.baseCurrency.code)?.value
+//            DateHelper.getSecondsAgo(record.timestamp * 1000) < latestRateFallbackThreshold * 60 -> rateManager.latestRates[record.coinCode]?.get(currencyManager.baseCurrency.code)?.value
```
Comment (rafaelekol) [inline app/src/main/java/io/horizontalsystems/bankwallet/core/factories/TransactionViewItemFactory.kt]: Do we need this commented code?

File: app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AuthManager.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (rafaelekol) [reply app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AuthManager.kt]: Maybe better would be to add TODO

[the remark this answers is not in the document]
Comment (abdrasulov) [reply app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AuthManager.kt]: Added TODO

File: app/src/main/java/io/horizontalsystems/bankwallet/core/factories/TransactionViewItemFactory.kt (hunk 3 above)
Comment (abdrasulov) [reply app/src/main/java/io/horizontalsystems/bankwallet/core/factories/TransactionViewItemFactory.kt]: inlined this method

File: app/src/main/java/io/horizontalsystems/bankwallet/SingleLiveEvent.kt (hunk 2 above)
Comment (abdrasulov) [reply app/src/main/java/io/horizontalsystems/bankwallet/SingleLiveEvent.kt]: inlined this method

File: app/src/main/java/io/horizontalsystems/bankwallet/core/factories/TransactionViewItemFactory.kt (hunk 3 above)
Comment (abdrasulov) [reply app/src/main/java/io/horizontalsystems/bankwallet/core/factories/TransactionViewItemFactory.kt]: removed

File: app/src/main/java/io/horizontalsystems/bankwallet/core/Interfaces.kt [hunk 4]
```diff
@@ -75,17 +76,11 @@ interface IKeyStoreSafeExecute {
...
-    fun createWords()
+    var backedUpSignal: PublishSubject<Unit>
+
     fun validate(words: List<String>)
-    fun restore(words: List<String>)
-    fun logout()
+    fun generateWords(): List<String>
 }
 
 enum class LogInState {
```
Comment (rafaelekol) [inline app/src/main/java/io/horizontalsystems/bankwallet/core/Interfaces.kt]: This Enum `LogInState` seems to be not used anymore

**Pull request merged**